File: management/analyze_listings.js
#!/usr/bin/env node
/**
 * Analyze listings and suggest standard assignments
 */

import { createClient } from '@supabase/supabase-js';

const supabaseUrl = 'https://haimjeaetrsaauitrhfy.supabase.co';
const supabaseKey = 'sb_secret_zRN1fTFOYnN7cEbEIfAP7A_YrEKBfI2';

const supabase = createClient(supabaseUrl, supabaseKey);

async function main() {
  console.log('🔍 Analyzing TSTR.site listings...\n');

  // Get categories
  const { data: categories, error: catError } = await supabase
    .from('categories')
    .select('id, name, slug')
    .order('name');

  if (catError) {
    console.error('Error fetching categories:', catError);
    return;
  }

  console.log('📊 Categories:');
  categories.forEach(cat => {
    console.log(`  - ${cat.name} (${cat.slug})`);
  });
  console.log();

  // Get listings by category
  const { data: listings, error: listError } = await supabase
    .from('listings')
    .select('id, business_name, category:category_id(id, name, slug), address')
    .eq('status', 'active')
    .order('business_name');

  if (listError) {
    console.error('Error fetching listings:', listError);
    return;
  }

  console.log(`📋 Total Active Listings: ${listings.length}\n`);

  // Group by category
  const byCategory = {};
  listings.forEach(listing => {
    const catName = listing.category?.name || 'Uncategorized';
    if (!byCategory[catName]) byCategory[catName] = [];
    byCategory[catName].push(listing);
  });

  console.log('📈 Listings by Category:');
  Object.entries(byCategory).forEach(([catName, catListings]) => {
    console.log(`\n  ${catName}: ${catListings.length} listings`);
    console.log('  ' + '-'.repeat(50));
    catListings.slice(0, 5).forEach(l => {
      console.log(`    • ${l.business_name}`);
      console.log(`      ${l.address || 'No address'}`);
    });
    if (catListings.length > 5) {
      console.log(`    ... and ${catListings.length - 5} more`);
    }
  });

  // Get standards
  const { data: standards, error: stdError } = await supabase
    .from('standards')
    .select('id, code, name, issuing_body, category:category_id(name)')
    .eq('is_active', true)
    .order('code');

  if (stdError) {
    console.error('Error fetching standards:', stdError);
    return;
  }

  console.log(`\n\n🎯 Available Standards: ${standards.length}`);
  
  // Group standards by category
  const stdByCategory = {};
  standards.forEach(std => {
    const catName = std.category?.name || 'General';
    if (!stdByCategory[catName]) stdByCategory[catName] = [];
    stdByCategory[catName].push(std);
  });

  console.log('\n📋 Standards by Category:');
  Object.entries(stdByCategory).forEach(([catName, catStandards]) => {
    console.log(`\n  ${catName}: ${catStandards.length} standards`);
    catStandards.forEach(s => {
      console.log(`    • ${s.code} - ${s.name}`);
    });
  });

  // Check existing capabilities
  const { data: capabilities, error: capError } = await supabase
    .from('listing_capabilities')
    .select('count');

  console.log(`\n\n📊 Current Capabilities: ${capabilities?.[0]?.count || 0}`);

  console.log('\n✅ Analysis complete!\n');
}

main().catch(console.error);

File: management/analyze_listings.py
#!/usr/bin/env python3
"""
Analyze listings and suggest standard assignments
"""

import requests
import json
from collections import defaultdict

SUPABASE_URL = "https://haimjeaetrsaauitrhfy.supabase.co"
SERVICE_ROLE_KEY = "sb_secret_zRN1fTFOYnN7cEbEIfAP7A_YrEKBfI2"

headers = {
    "apikey": SERVICE_ROLE_KEY,
    "Authorization": f"Bearer {SERVICE_ROLE_KEY}",
    "Content-Type": "application/json"
}

def fetch_data(table, select="*", filters=None):
    """Fetch data from Supabase"""
    url = f"{SUPABASE_URL}/rest/v1/{table}"
    params = {"select": select}
    if filters:
        params.update(filters)
    
    response = requests.get(url, headers=headers, params=params)
    if response.status_code == 200:
        return response.json()
    else:
        print(f"Error fetching {table}: {response.text}")
        return None

def main():
    print("🔍 Analyzing TSTR.site listings...\n")

    # Get categories
    categories = fetch_data("categories", "id,name,slug", {"order": "name"})
    if not categories:
        return

    print("📊 Categories:")
    cat_map = {}
    for cat in categories:
        print(f"  - {cat['name']} ({cat['slug']})")
        cat_map[cat['id']] = cat['name']
    print()

    # Get listings
    listings = fetch_data(
        "listings",
        "id,business_name,category_id,address",
        {"status": "eq.active", "order": "business_name"}
    )
    if not listings:
        return

    print(f"📋 Total Active Listings: {len(listings)}\n")

    # Group by category
    by_category = defaultdict(list)
    for listing in listings:
        cat_name = cat_map.get(listing['category_id'], 'Uncategorized')
        by_category[cat_name].append(listing)

    print("📈 Listings by Category:")
    for cat_name, cat_listings in sorted(by_category.items()):
        print(f"\n  {cat_name}: {len(cat_listings)} listings")
        print("  " + "-" * 50)
        for listing in cat_listings[:5]:
            print(f"    • {listing['business_name']}")
            addr = listing.get('address', 'No address')
            if addr and len(addr) > 60:
                addr = addr[:57] + "..."
            print(f"      {addr}")
        if len(cat_listings) > 5:
            print(f"    ... and {len(cat_listings) - 5} more")

    # Get standards
    standards = fetch_data(
        "standards",
        "id,code,name,issuing_body,category_id",
        {"is_active": "eq.true", "order": "code"}
    )
    if not standards:
        return

    print(f"\n\n🎯 Available Standards: {len(standards)}")

    # Group standards by category
    std_by_category = defaultdict(list)
    for std in standards:
        cat_name = cat_map.get(std['category_id'], 'General')
        std_by_category[cat_name].append(std)

    print("\n📋 Standards by Category:")
    for cat_name, cat_standards in sorted(std_by_category.items()):
        print(f"\n  {cat_name}: {len(cat_standards)} standards")
        for std in cat_standards:
            print(f"    • {std['code']} - {std['name']}")

    # Check existing capabilities
    capabilities = fetch_data("listing_capabilities", "count")
    cap_count = capabilities[0]['count'] if capabilities and len(capabilities) > 0 else 0
    print(f"\n\n📊 Current Capabilities: {cap_count}")

    # Save data for assignment script
    data = {
        'categories': cat_map,
        'listings': listings,
        'standards': standards,
        'by_category': {k: [listing['id'] for listing in v] for k, v in by_category.items()}
    }
    
    with open('/tmp/listings_data.json', 'w') as f:
        json.dump(data, f, indent=2)
    
    print("\n💾 Data saved to /tmp/listings_data.json")
    print("\n✅ Analysis complete!\n")

if __name__ == "__main__":
    main()
